File: app/src/main/java/com/example/bestreminder/ReminderPreferences.java
package com.example.bestreminder;

import android.content.Context;
import android.content.SharedPreferences;

final class ReminderPreferences {
    static final int DEFAULT_HOUR = 9;
    static final int DEFAULT_MINUTE = 0;

    private static final String FILE_NAME = "best_reminder_preferences";
    private static final String KEY_ENABLED = "enabled";
    private static final String KEY_HOUR = "hour";
    private static final String KEY_MINUTE = "minute";

    private ReminderPreferences() {
    }

    static boolean isEnabled(Context context) {
        return preferences(context).getBoolean(KEY_ENABLED, false);
    }

    static int getHour(Context context) {
        return preferences(context).getInt(KEY_HOUR, DEFAULT_HOUR);
    }

    static int getMinute(Context context) {
        return preferences(context).getInt(KEY_MINUTE, DEFAULT_MINUTE);
    }

    static void save(Context context, boolean enabled, int hour, int minute) {
        preferences(context)
                .edit()
                .putBoolean(KEY_ENABLED, enabled)
                .putInt(KEY_HOUR, hour)
                .putInt(KEY_MINUTE, minute)
                .apply();
    }

    static void setEnabled(Context context, boolean enabled) {
        preferences(context).edit().putBoolean(KEY_ENABLED, enabled).apply();
    }

    private static SharedPreferences preferences(Context context) {
        return context.getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE);
    }
}

File: app/src/main/java/com/example/thebestreminder/ReminderPreferences.kt
package com.example.thebestreminder

import android.content.Context

data class ReminderSettings(
    val hour: Int,
    val minute: Int,
    val enabled: Boolean,
)

class ReminderPreferences(context: Context) {
    private val preferences = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)

    fun loadSettings(): ReminderSettings =
        ReminderSettings(
            hour = preferences.getInt(KEY_HOUR, DEFAULT_HOUR),
            minute = preferences.getInt(KEY_MINUTE, DEFAULT_MINUTE),
            enabled = preferences.getBoolean(KEY_ENABLED, false),
        )

    fun saveTime(hour: Int, minute: Int) {
        preferences.edit()
            .putInt(KEY_HOUR, hour)
            .putInt(KEY_MINUTE, minute)
            .apply()
    }

    fun setEnabled(enabled: Boolean) {
        preferences.edit()
            .putBoolean(KEY_ENABLED, enabled)
            .apply()
    }

    companion object {
        private const val PREFS_NAME = "the_best_reminder_prefs"
        private const val KEY_HOUR = "reminder_hour"
        private const val KEY_MINUTE = "reminder_minute"
        private const val KEY_ENABLED = "reminder_enabled"

        private const val DEFAULT_HOUR = 9
        private const val DEFAULT_MINUTE = 0
    }
}
